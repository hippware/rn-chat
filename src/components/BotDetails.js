import React from 'react'
import {
    View,
    TouchableWithoutFeedback,
    Text,
    ScrollView,
    Animated,
    Alert,
    TouchableOpacity,
    Image
} from 'react-native'
import { Actions } from 'react-native-router-native'
import Screen from './Screen'
import botFactory from '../factory/botFactory'
import Map from './Map'
import { Annotation } from 'react-native-mapbox-gl'
import GradientHeader from './GradientHeader'
import { k, width, height, defaultCover } from './Global'
import BotAvatar from './BotAvatar'
import Avatar from './Avatar'
import { observer } from 'mobx-react/native'
import { observable } from 'mobx'
import botStore from '../store/botStore'
import location from '../store/locationStore'
import Bot, { VISIBILITY_PUBLIC, VISIBILITY_OWNER, LOCATION, NOTE, IMAGE } from '../model/Bot'
import autobind from 'autobind-decorator'
import statem from '../../gen/state'
import PhotoGrid from './PhotoGrid'
import model from '../model/model'
import { when } from 'mobx'
import BotNavBar from './BotNavBar'

const DOUBLE_PRESS_DELAY = 300
function Header (props) {
    return <View style={{
        backgroundColor: 'rgba(255,255,255,0.85)',
        flexDirection: 'row',
        height: 41 * k,
        shadowOffset: {height: 1, width: 0},
        shadowRadius: 2,
        shadowOpacity: 0.12,
    }}>
        <View style={{flex: 1, justifyContent: 'center'}}>
            <Text numberOfLines={1} style={{
                paddingLeft: 26 * k,
                fontSize: 14 * k,
                color: 'rgb(63,50,77)',
                fontFamily: 'Roboto-Regular'
            }}>{props.children}</Text>
        </View>
    </View>
}

@autobind
@observer
export default class extends React.Component {

    constructor (props) {
        super(props)
        this.loading = false
        this.state = {
            top: new Animated.Value(this.props.fullMap ? height : 0),
            fullMap: !!this.props.fullMap,
            fadeAnim: new Animated.Value(0),
            showNavBar: true,
            navBarHeight: new Animated.Value(70),
        }
    }

    async loadMoreImages () {
        if (botStore.bot && botStore.bot.imagesCount && botStore.bot.images.length && botStore.bot.imagesCount > botStore.bot.images.length) {
            if (!this.loading) {
                this.loading = true
                console.log('LOAD MORE IMAGES', botStore.bot.images[botStore.bot.images.length - 1].item)
                await botStore.loadImages(botStore.bot.images[botStore.bot.images.length - 1].item)
                this.loading = false
            }
        }
    }

    onScrollStart () {
        // display 'no more images'
        if (botStore.bot.imagesCount > 0 && botStore.bot.imagesCount === botStore.bot.images.length) {
            this.setState({showNoMoreImages: true})
        }
    }

    onScrollEnd (event) {
        // load more images
        if (!this.state.showNavBar) {
            console.log('SCROLL END!', botStore.bot.imagesCount, botStore.bot.images.length)
            // this.setState({showNavBar: true})
            // Animated.timing(
            //   this.state.navBarHeight,
            //   {toValue: 70}
            // ).start();
        }
        this.setState({showNoMoreImages: false})

    }

    onScroll (event) {
        if (event.nativeEvent.contentOffset.y + height + 200 >= event.nativeEvent.contentSize.height) {
            this.loadMoreImages()
        }
        // if (this.state.showNavBar){
        //   this.setState({showNavBar: false})
        //   Animated.timing(
        //     this.state.navBarHeight,
        //     {toValue: 0}
        //   ).start();
        // }
        // switch nav bar is scroll position is below threshold
        // const y = event.nativeEvent.contentOffset.y;
        // const limit = 50 * k;
        // const upper = 240*k;
        // if (y > limit) {
        //   Animated.timing(          // Uses easing functions
        //     this.state.fadeAnim,    // The value to drive
        //     {toValue: Math.max(0, 1 - y/upper)}            // Configuration
        //   ).start();
        //   if (!this.state.showNavBar) {
        //     this.setState({showNavBar: true});
        //   }
        // } else {
        //   Animated.timing(          // Uses easing functions
        //     this.state.fadeAnim,    // The value to drive
        //     {toValue: Math.min(1, 1 - y/upper)}            // Configuration
        //   ).start();
        //   if (this.state.showNavBar) {
        //     this.setState({showNavBar: false});
        //   }
        // }
    }

    async componentWillMount () {
        if (this.props.item) {
            botStore.bot = botFactory.create({id: this.props.item})
            when(() => model.connected, botStore.load)
        }
    }

    hideFullMap () {
        this.setState({fullMap: true})
        Animated.timing(
            this.state.top,
            {toValue: 0}
        ).start()
        Actions.refresh({})
    }

    showFullMap () {
        this.setState({fullMap: true})
        Animated.timing(
            this.state.top,
            {toValue: height}
        ).start()
        Actions.refresh({leftButton: {icon: require('../../images/iconClose.png'), onPress: this.hideFullMap}})
    }

    onLayout (event) {
        var layout = event.nativeEvent.layout
        this.setState({currentScreenWidth: layout.width, currentScreenHeight: layout.height})
    }

    unsubscribe () {
        Alert.alert(null, 'Are you sure you want to unsubscribe?', [
            {text: 'Cancel', style: 'cancel'},
            {text: 'Unsubscribe', style: 'destructive', onPress: () => botStore.unsubscribe()}
        ])
    }

    subscribe () {
        botStore.subscribe()
        // do animation
        this.setState({fadeAnim: new Animated.Value(1)})
        setTimeout(() => {
            Animated.timing(
                this.state.fadeAnim,
                {toValue: 0}
            ).start()

        }, 500)
    }

// ...

    /**
     * Double Press recognition
     * @param  {Event} e
     */
    handleImagePress (e) {
        const now = new Date().getTime()

        if (this.lastImagePress && (now - this.lastImagePress) < DOUBLE_PRESS_DELAY) {
            delete this.lastImagePress
            this.handleImageDoublePress(e)
        }
        else {
            this.lastImagePress = now
        }
    }

    handleImageDoublePress (e) {
        const bot = botStore.bot
        if (!bot.isSubscribed) {
            this.subscribe()
        }
    }

    render () {
        const bot = botStore.bot
        if (!bot) {
            console.log('ERROR: No bot defined', this.props.item)
            return <Screen/>
        }
        const isDay = location.isDay
        const isOwn = !bot.owner || bot.owner.isOwn
        const coef = bot.image && bot.image.width ? (width - 34 * k) / bot.image.width : 0
        const profile = bot.owner
        if (!profile) {
            console.log('ERROR: NO BOT PROFILE!')
            return <Screen/>
        }
        const source = bot.image && bot.image.source
        return <View style={{flex: 1, backgroundColor: location.isDay ? 'white' : 'rgba(49,37,62,0.90)'}}>
            <ScrollView style={{paddingTop: 70 * k}} onScrollEndDrag={this.onScrollEnd}
                        onScrollBeginDrag={this.onScrollStart} onScroll={this.onScroll} scrollEventThrottle={1}>
                <View style={{width: 375 * k, height: 275 * k}}>
                    <TouchableWithoutFeedback onPress={this.handleImagePress}>
                        <Image style={{width: 375 * k, height: 275 * k}}
                               source={source || defaultCover[bot.coverColor % 4]}/>
                    </TouchableWithoutFeedback>
                    {isOwn && <TouchableOpacity onPress={() => statem.logged.botEdit({item: bot.id})}
                                                style={{
                                                    borderRadius: 2,
                                                    backgroundColor: 'rgba(255,255,255,0.75)',
                                                    position: 'absolute',
                                                    justifyContent: 'center',
                                                    alignItems: 'center',
                                                    bottom: 20 * k,
                                                    right: 20 * k,
                                                    height: 30 * k,
                                                    width: 36 * k
                                                }}>
                        <Image source={require('../../images/iconEditBot.png')}/>
                    </TouchableOpacity>}
                    <Animated.View pointerEvents="none" style={{
                        width: 375 * k, height: 275 * k, opacity: this.state.fadeAnim,
                        position: 'absolute', justifyContent: 'center', alignItems: 'center'
                    }}>
                        <Image source={require('../../images/iconBotAdded.png')}/>
                    </Animated.View>
                </View>
                <View style={{paddingTop: 15 * k, paddingLeft: 20 * k, paddingRight: 20 * k}}>
                    {!isOwn && !bot.isSubscribed && <TouchableOpacity onPress={this.subscribe} style={{
                        height: 40 * k,
                        borderWidth: 0,
                        backgroundColor: 'rgb(254,92,108)',
                        borderRadius: 2 * k,
                        justifyContent: 'center',
                        alignItems: 'center'
                    }}>
                        <Text
                            style={{fontSize: 11 * k, letterSpacing: 0.5, fontFamily: 'Roboto-Medium', color: 'white'}}>ADD
                            BOT</Text>
                    </TouchableOpacity>}
                    {!isOwn && !!bot.isSubscribed && <TouchableOpacity onPress={this.unsubscribe} style={{
                        height: 40 * k,
                        flexDirection: 'row',
                        justifyContent: 'center',
                        alignItems: 'center',
                        borderWidth: 0,
                        backgroundColor: 'rgb(228,228,228)',
                        borderRadius: 2 * k
                    }}>
                        <View style={{padding: 10 * k}}><Image source={require('../../images/iconCheckBotAdded.png')}/></View>
                        <Text style={{
                            fontSize: 11 * k,
                            letterSpacing: 0.5,
                            fontFamily: 'Roboto-Medium',
                            color: 'rgb(99,62,90)'
                        }}>BOT ADDED</Text>
                    </TouchableOpacity>}
                </View>
                <View style={{
                    paddingTop: 15 * k,
                    paddingBottom: 15 * k,
                    paddingLeft: 20 * k,
                    paddingRight: 20 * k,
                    flexDirection: 'row',
                    alignItems: 'center'
                }}>
                    <View style={{paddingRight: 11 * k}}><Avatar size={36} profile={profile}
                                                                 source={profile.avatar && profile.avatar.source}
                                                                 title={profile.displayName} isDay={location.isDay}
                                                                 disableStatus borderWidth={0}/></View>
                    <View style={{flex: 1}}><Text style={{
                        fontFamily: 'Roboto-Italic',
                        fontSize: 13,
                        letterSpacing: -0.1,
                        color: 'rgb(114,100,109)'
                    }}>@{profile.handle}</Text></View>
                    {location.location && bot.location && <View>
                        <Image source={require('../../images/buttonViewMapBG.png')}/>
                        <TouchableOpacity onPress={statem.botDetails.map} style={{
                            position: 'absolute',
                            top: 0,
                            bottom: 0,
                            right: 0,
                            left: 0,
                            justifyContent: 'center',
                            alignItems: 'center',
                            flexDirection: 'row',
                            backgroundColor: 'transparent'
                        }}>
                            <View style={{paddingRight: 5}}><Image
                                source={require('../../images/iconBotLocation.png')}/></View>
                            <Text style={{fontFamily: 'Roboto-Regular', fontSize: 13, color: 'rgb(63,50,77)'}}>
                                {location.distanceToString(
                                    location.distance(location.location.latitude, location.location.longitude, bot.location.latitude, bot.location.longitude)
                                )}
                            </Text>
                        </TouchableOpacity>
                    </View>}
                </View>
                {!!bot.description && <View style={{paddingLeft: 20 * k, paddingRight: 20 * k, paddingBottom: 15 * k}}>
                    <Text numberOfLines={0} style={{
                        fontFamily: 'Roboto-Light',
                        fontSize: 15,
                        color: location.isDay ? 'rgb(63,50,77)' : 'white'
                    }}>{bot.description}</Text>
                </View>
                }
                {!isOwn && !bot.imagesCount && <View style={{
                    height: 201 * k,
                    backgroundColor: 'rgb(242,243,245)',
                    justifyContent: 'center',
                    alignItems: 'center'
                }}>
                    <Image source={require('../../images/attachPhotoGray.png')}/>
                    <Text style={{fontFamily: 'Roboto-Regular', fontSize: 15, color: 'rgb(186,186,186)'}}>No photos
                        added</Text>
                </View>}
                <PhotoGrid isOwn={isOwn} images={bot.images} onAdd={statem.botDetails.addPhoto}
                           onView={index => statem.botDetails.editPhotos({index})}/>
                {this.state.showNoMoreImages &&
                <View style={{paddingTop: 10, alignItems: 'center', paddingBottom: 21}}><Image
                    source={require('../../images/graphicEndPhotos.png')}/></View>}
            </ScrollView>
            {this.state.showNavBar && <BotNavBar bot={bot}/>}
        </View>
    }
}