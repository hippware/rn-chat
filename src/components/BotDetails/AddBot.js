// @flow

import React from 'react';
import {Text, TouchableOpacity, StyleSheet, Image} from 'react-native';
import {colors} from '../../constants';
import {k} from '../../globals';
import {observer} from 'mobx-react/native';

export default observer(({subscribe, unsubscribe, isSubscribed}) => {
  let onPress, buttonStyle, image, text, textStyle;
  if (isSubscribed) {
    onPress = unsubscribe;
    buttonStyle = [styles.addBotButton, {backgroundColor: colors.WHITE, borderColor: colors.GREY}];
    image = require('../../../images/iconCheckBotAdded.png');
    text = 'SAVED';
    textStyle = [styles.addBotText, {color: colors.PURPLE}];
  } else {
    onPress = subscribe;
    buttonStyle = styles.addBotButton;
    image = require('../../../images/saveIcon.png');
    text = 'SAVE';
    textStyle = styles.addBotText;
  }
  return (
    <TouchableOpacity onPress={onPress} style={buttonStyle}>
      <Image source={image} style={styles.addBotIcon} resizeMode='contain' />
      <Text style={textStyle}>
        {text}
      </Text>
    </TouchableOpacity>
  );
});

const styles = StyleSheet.create({
  addBotButton: {
    flexDirection: 'row',
    height: 40 * k,
    marginTop: -20 * k,
    width: 150 * k,
    alignSelf: 'center',
    backgroundColor: colors.PINK,
    borderRadius: 5 * k,
    borderColor: colors.PINK,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  addBotText: {
    fontSize: 15,
    letterSpacing: 0.5,
    fontFamily: 'Roboto-Medium',
    color: 'white',
    marginLeft: 5 * k,
  },
});
