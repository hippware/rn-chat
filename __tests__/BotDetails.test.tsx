import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import BotDetails from '../src/components/BotDetails/BotDetails'
import {Provider} from 'mobx-react/native'
import mockStore from './utils/mockStore'

describe('BotDetails', () => {
  it('renders with no data', () => {
    const toRender = (
      <Provider {...mockStore}>
        <BotDetails botId={'1234'} isNew={false} navigation={{state: {params: {isNew: false}}}} />
      </Provider>
    )
    const tree = renderer.create(toRender).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
