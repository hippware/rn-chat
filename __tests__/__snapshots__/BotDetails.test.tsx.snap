// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotDetails renders with no data 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <R2RTouchThroughWrapper
    style={
      Object {
        "height": 1334,
        "width": 750,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListFooterComponent={[Function]}
      ListHeaderComponent={
        <DraggablePopupListHeader
          inner={
            <inject-BotDetailsHeader-with-notificationStore-analytics-locationStore
              analytics={
                Object {
                  "track": [MockFunction],
                }
              }
              bot={
                Object {
                  "address": "",
                  "addressData": Object {
                    "city": "",
                    "country": "",
                    "county": "",
                    "state": "",
                  },
                  "description": "",
                  "followersSize": 0,
                  "guest": false,
                  "guestsSize": 0,
                  "icon": "",
                  "id": "1234",
                  "image": null,
                  "invitation": null,
                  "isSubscribed": false,
                  "location": null,
                  "owner": null,
                  "radius": 100,
                  "server": null,
                  "title": null,
                  "totalItems": 0,
                  "visitor": false,
                  "visitors": Object {
                    "count": null,
                    "cursor": null,
                    "result": Array [],
                  },
                  "visitorsSize": 0,
                }
              }
              botId="1234"
              homeStore={Object {}}
              isNew={false}
              navigation={
                Object {
                  "state": Object {
                    "params": Object {
                      "isNew": false,
                    },
                  },
                }
              }
              notificationStore={Object {}}
              wocky={
                Object {
                  "_acceptBotInvitation": [MockFunction],
                  "_addMessage": [MockFunction],
                  "_block": [MockFunction],
                  "_follow": [MockFunction],
                  "_hideUser": [MockFunction],
                  "_inviteBot": [MockFunction],
                  "_loadBotGuests": [MockFunction],
                  "_loadBotPosts": [MockFunction],
                  "_loadBotSubscribers": [MockFunction],
                  "_loadBotVisitors": [MockFunction],
                  "_loadGeofenceBots": [MockFunction],
                  "_loadNewNotifications": [MockFunction],
                  "_loadNotifications": [MockFunction],
                  "_loadOwnBots": [MockFunction],
                  "_loadRelations": [MockFunction],
                  "_loadSubscribedBots": [MockFunction],
                  "_onBotVisitor": [MockFunction],
                  "_onNotification": [MockFunction],
                  "_publishBotPost": [MockFunction],
                  "_registerReferences": [MockFunction],
                  "_removeBotPost": [MockFunction],
                  "_removeUpload": [MockFunction],
                  "_requestProfiles": [MockFunction],
                  "_requestUpload": [MockFunction],
                  "_sendMessage": [MockFunction],
                  "_shareBot": [MockFunction],
                  "_subscribeBot": [MockFunction],
                  "_unblock": [MockFunction],
                  "_unfollow": [MockFunction],
                  "_unsubscribeBot": [MockFunction],
                  "_updateBot": [MockFunction],
                  "_updateProfile": [MockFunction],
                  "activeBots": Object {},
                  "addRosterItem": [MockFunction],
                  "afterCreate": [MockFunction],
                  "all": Object {},
                  "blocked": Object {},
                  "bots": Object {},
                  "chats": Object {},
                  "clearCache": [MockFunction],
                  "connected": true,
                  "connecting": false,
                  "create": [MockFunction],
                  "createBot": [MockFunction],
                  "createChat": [MockFunction],
                  "createProfile": [MockFunction],
                  "deleteBot": [MockFunction],
                  "disablePush": [MockFunction],
                  "disconnect": [MockFunction],
                  "disposeReactions": [MockFunction],
                  "downloadFile": [MockFunction],
                  "downloadTROS": [MockFunction],
                  "downloadThumbnail": [MockFunction],
                  "downloadURL": [MockFunction],
                  "enablePush": [MockFunction],
                  "files": Object {},
                  "followed": Object {},
                  "followers": Object {},
                  "friends": Object {},
                  "geofenceBots": Object {},
                  "getBot": [Function],
                  "getLocationsVisited": [MockFunction],
                  "getProfile": [MockFunction],
                  "hasUnreadNotifications": false,
                  "host": "host",
                  "id": "wocky",
                  "load": [MockFunction],
                  "loadBot": [MockFunction] {
                    "calls": Array [
                      Array [
                        "1234",
                        undefined,
                      ],
                    ],
                    "results": Array [
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                    ],
                  },
                  "loadChat": [MockFunction],
                  "loadChats": [MockFunction],
                  "loadLocalBots": [MockFunction],
                  "loadProfile": [MockFunction],
                  "login": [MockFunction],
                  "logout": [MockFunction],
                  "lookup": [MockFunction],
                  "map": [MockFunction],
                  "newFollowers": Object {},
                  "notifications": Object {},
                  "password": "password",
                  "profile": Object {
                    "email": "eric.kirkham@gmail.com",
                    "firstName": "eric",
                    "handle": "jerkham",
                    "lastName": "kirkham",
                    "loaded": true,
                    "updateError": "",
                  },
                  "profiles": Object {},
                  "register": [MockFunction],
                  "remove": [MockFunction],
                  "removeBot": [MockFunction],
                  "requestRoster": [MockFunction],
                  "roster": Object {},
                  "searchUsers": [MockFunction],
                  "service": [MockFunction],
                  "sessionCount": 0,
                  "setLocation": [MockFunction],
                  "setSessionCount": [MockFunction],
                  "sortedRoster": Array [],
                  "startReactions": [MockFunction],
                  "testRegister": [MockFunction],
                  "transport": Object {},
                  "userInviteMakeCode": [MockFunction],
                  "userInviteRedeemCode": [MockFunction],
                  "username": "username",
                  "viewNotifications": [MockFunction],
                }
              }
            />
          }
        />
      }
      bounces={false}
      contentContainerStyle={
        Object {
          "flexGrow": 1,
          "paddingBottom": 50,
        }
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={8}
      keyExtractor={[Function]}
      keyboardDismissMode="on-drag"
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <R2RTouchThroughView
              style={
                Object {
                  "height": 667,
                  "width": 750,
                }
              }
            />
            <View
              style={
                Object {
                  "paddingTop": 50,
                }
              }
            >
              <Image
                resizeMode="stretch"
                source={
                  Object {
                    "testUri": "../../../images/bottomPopup.png",
                  }
                }
                style={
                  Object {
                    "position": "absolute",
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "flex": 1,
                    "marginTop": 20,
                    "paddingHorizontal": 40,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "color": "rgb(63,50,77)",
                            "fontFamily": "Roboto-Regular",
                            "fontSize": 20,
                          },
                          Object {
                            "textAlign": "center",
                            "width": "75%",
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginTop": 20,
                      }
                    }
                  >
                    <BVLinearGradient
                      colors={
                        Array [
                          "rgb(242,68,191)",
                          "rgb(254,110,98)",
                          "rgb(254,92,108)",
                        ]
                      }
                      end={
                        Object {
                          "x": 1,
                          "y": 0.5,
                        }
                      }
                      start={
                        Object {
                          "x": 0,
                          "y": 0.5,
                        }
                      }
                      style={
                        Object {
                          "borderRadius": 5,
                          "marginRight": 10,
                          "paddingHorizontal": 20,
                          "paddingVertical": 6,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "rgb(255,255,255)",
                              "fontFamily": "Roboto-Medium",
                              "fontSize": 12,
                            },
                            undefined,
                          ]
                        }
                      >
                        
                      </Text>
                    </BVLinearGradient>
                    <BVLinearGradient
                      colors={
                        Array [
                          "rgb(242,68,191)",
                          "rgb(254,110,98)",
                          "rgb(254,92,108)",
                        ]
                      }
                      end={
                        Object {
                          "x": 1,
                          "y": 0.5,
                        }
                      }
                      start={
                        Object {
                          "x": 0,
                          "y": 0.5,
                        }
                      }
                      style={
                        Object {
                          "borderRadius": 5,
                          "marginRight": 10,
                          "paddingHorizontal": 20,
                          "paddingVertical": 6,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "rgb(255,255,255)",
                              "fontFamily": "Roboto-Medium",
                              "fontSize": 12,
                            },
                            undefined,
                          ]
                        }
                      />
                    </BVLinearGradient>
                  </View>
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "height": 17,
                          }
                        }
                      />
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "rgb(222,222,222)",
                              "height": 2,
                              "marginHorizontal": 50,
                              "marginVertical": 40,
                            },
                            Object {
                              "marginHorizontal": 5,
                              "marginTop": 0,
                              "width": "100%",
                            },
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "rgb(255,255,255)",
                          "flexDirection": "row",
                          "marginBottom": 10,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "height": 160,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgb(222,222,222)",
                            "height": 2,
                            "marginHorizontal": 50,
                            "marginVertical": 40,
                          },
                          Object {
                            "marginHorizontal": 5,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 1,
            "borderColor": "rgb(212,212,212)",
            "elevation": 1,
            "flex": 1,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "space-between",
            "paddingHorizontal": 10,
            "paddingTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "width": 23,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/iconBackGrayNew.png",
                }
              }
              style={
                Object {
                  "height": 21,
                  "marginLeft": 10,
                  "tintColor": "rgb(117,117,117)",
                  "width": 13,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginHorizontal": 16,
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "rgb(63,50,77)",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "color": "rgb(63,50,77)",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 32,
                      "fontWeight": undefined,
                      "letterSpacing": 0.5,
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ],
                ]
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "rgb(63,50,77)",
                    "fontFamily": "Roboto-Light",
                    "fontSize": 12,
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginRight": 20,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/shareIcon.png",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </R2RTouchThroughWrapper>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255,255,255)",
          "borderColor": "rgb(212,212,212)",
          "borderTopWidth": 1,
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "white",
              "flexDirection": "row",
              "padding": 15,
            },
          ]
        }
      >
        <View
          accessible={true}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 15,
              "right": 15,
              "top": 15,
            }
          }
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 0,
              "opacity": 1,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../images/attachPhoto.png",
              }
            }
            style={
              Object {
                "height": 21,
              }
            }
          />
        </View>
        <TextInput
          allowFontScaling={true}
          enablesReturnKeyAutomatically={true}
          maxLength={5000}
          multiline={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Add a comment"
          placeholderTextColor="rgb(155,155,155)"
          returnKeyType="default"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "flex": 1,
                "fontFamily": "Roboto-Regular",
                "fontSize": 14,
                "margin": 0,
                "paddingHorizontal": 11,
                "paddingTop": 0,
              },
              Object {
                "color": "rgb(63,50,77)",
              },
              Object {
                "height": "auto",
              },
            ]
          }
          value=""
        />
        <View
          accessible={true}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 15,
              "right": 15,
              "top": 15,
            }
          }
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(212,212,212)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 16,
                },
                undefined,
              ]
            }
          >
            Post
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </View>
  </View>
</View>
`;
